### PR TITLE
Add generated build files in git. Helps review builder changes

### PR DESCRIPTION
**Pull request by bogdandrutu**
Ensure that the generated files are up to date.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

File: cmd/otelcorecol/main.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+func main() {
+	factories, err := components()
+	if err != nil {
+		log.Fatalf("failed to build components: %v", err)
+	}
+
+	info := component.BuildInfo{
+		Command:  "otelcorecol",
+		Description: "Local OpenTelemetry Collector binary, testing only.",
+		Version:  "0.41.0-dev",
```
Comment (codeboten) [inline cmd/otelcorecol/main.go:24]: Thoughts on how to keep this version in sync as the version gets updated?

**Review by codeboten — COMMENTED** [review]
Please run `make gendependabot` as well.

**Comment by bogdandrutu** [comment]
@codeboten `dependabot` should not update anything since files are auto-generated including `go.mod`

File: cmd/otelcorecol/main.go (hunk 1 above)
Comment (bogdandrutu) [reply cmd/otelcorecol/main.go:24]: This is generated from the builder-config.yaml, not sure how to keep that up to date.

**Review by codeboten — APPROVED** [review]

**Pull request merged**